openedx/features/enterprise_support/api.py

```diff
@@ -43,6 +41,39 @@ class EnterpriseApiException(Exception):
     pass
 
 
+class ConsentApiClient(object):
+
+    def __init__(self):
+        self.user = User.objects.get(username=settings.ENTERPRISE_SERVICE_WORKER_USERNAME)
+        jwt = JwtBuilder(self.user).build_token([])
+        self.client = EdxRestApiClient(
+            configuration_helpers.get_value('CONSENT_API_URL', settings.CONSENT_API_URL),
+            jwt=jwt,
+            append_slash=False,
+        )
+        self.consent_endpoint = self.client.data_sharing_consent
+
+    def revoke_consent(self, **kwargs):
+        return self.consent_endpoint.delete(**kwargs)
+
+    def provide_consent(self, **kwargs):
+        return self.consent_endpoint.post(kwargs)
+
+    def consent_needed(self, enrollment_exists=False, **kwargs):
+        from slumber.exceptions import HttpNotFoundError
+        try:
+            return self.consent_endpoint.get(**kwargs)['consent_required']
+        except HttpNotFoundError as e:
+            received_response = self.consent_endpoint._try_to_serialize_response(e.response)['consent_required']
+            if enrollment_exists:
```

Comment:
The case in which that would happen would be if the user had, independently of the EnterpriseCustomer context, enrolled in a course on their own. This could come up either due to a user having an existing account with courses when they made the link, by a change in the scope of the EnterpriseCustomer's catalog, or by the user specifically opting out of the Enterprise-context enrollment flow by browsing in the LMS to find the course.